cmd/deepsource/main.go

```diff
@@ -0,0 +1,28 @@
+package main
+
+import (
+	"runtime/debug"
+
+	"github.com/deepsourcelabs/cli/pkg/cmd"
+)
+
+var Version = "development"
+
+// Date is dynamically set at build time in the Makefile.
+var Date = "YYYY-MM-DD" // YYYY-MM-DD
+
+func main() {
+
+	buildVersion, buildDate := getBuildInfo()
+
+	cmd.Execute(buildVersion, buildDate)
+}
+
+func getBuildInfo() (string, string) {
+
+	if info, ok := debug.ReadBuildInfo(); ok && info.Main.Version != "(devel)" {
+		Version = info.Main.Version
+	}
+
+	return Version, Date
+}
```

Comment:
Why do we need to do this in the main.go?

Reply:
It can be done in any other file but since we set the version of main package in the build command (`go build -ldflags="-X 'main.Version=v1.0.0' -X 'main.Date=$(date)'"`), thus it to be there in `main` package. 

Should we make a different package (eg: `build`) and set the version and Date in it? Eg: `-ldflags="-X 'build.Version=v1.0.0' -X 'build.Date=$(date)'"`)`. In that case, we can move this logic to the build package instead of putting it in `main`.